lib/SILOptimizer/Utils/ValueLifetime.cpp

```diff
@@ -190,7 +197,8 @@ bool ValueLifetimeAnalysis::computeFrontier(Frontier &frontier, Mode mode,
 
     for (unsigned i = 0, e = succBlocks.size(); i != e; ++i) {
       if (unhandledFrontierBlocks.count(succBlocks[i])) {
-        assert(isCriticalEdge(term, i) && "actually not a critical edge?");
+        assert((isCriticalEdge(term, i) || userSet.count(term)) &&
```

Comment:
Can you explain this to me? Why is this change needed? Does userSet.count(term) mean a critical edge?

Reply:
This is a bug fix.
```
/// An edge is also considered as "critical" if it has a single precedessor
/// but the predecessor's terminal instruction is a user of the value.
///
```